crates/net/downloaders/src/bodies/task.rs

```diff
@@ -14,14 +14,18 @@ use std::{
     task::{ready, Context, Poll},
 };
 use tokio::sync::{mpsc, mpsc::UnboundedSender};
-use tokio_stream::wrappers::UnboundedReceiverStream;
+use tokio_stream::wrappers::{ReceiverStream, UnboundedReceiverStream};
+use tokio_util::sync::PollSender;
+
+/// The maximum number of [BodyDownloaderResult]s to hold in the buffer.
+pub const BODIES_TASK_BUFFER_SIZE: usize = 8;
```

Comment:
this is likely too high, 
we should also consider reducing the stream batch size from 10k because this can be quite big if blocks are big